osu.Game/Screens/OnlinePlay/Components/StatusColouredContainer.cs

```diff
@@ -29,18 +29,27 @@ protected override void LoadComplete()
             base.LoadComplete();
 
             room.PropertyChanged += onRoomPropertyChanged;
+
+            Scheduler.AddDelayed(updateRoomStatus, 5000, true);
```

Comment:
Just to confirm intent - are these schedules supposed to handle the passage of time in relation to `EndDate` in order to correctly transition to ended state?

Reply:
Correct. I thought of doing something similar to `DrawableDate` with a dynamic reschedule time but I don't think this needs to be accurate. Maybe in the future this will be messaged from the multiplayer server anyway...

Reply:
I think this is a bit weird.

Depending on the load time of individual components, the 5 second poll may occur at different times, causing some elements to update at different times, even on the same room display.

I don't have an immediate better solution in mind though.

Reply:
In general I expect these schedules to be triggered on the same frame for the same `DrawableRoom`. For different `DrawableRooms` - sure they can trigger at different times, but they're created at different times anyway so I don't see that as an issue.

As I said the future solution here is realtime updates from the multiplayer server.

Reply:
Could we use this for now?

```diff
diff --git a/osu.Game/Screens/OnlinePlay/Components/StatusColouredContainer.cs b/osu.Game/Screens/OnlinePlay/Components/StatusColouredContainer.cs
index a811ee3371..379dc65bcd 100644
--- a/osu.Game/Screens/OnlinePlay/Components/StatusColouredContainer.cs
+++ b/osu.Game/Screens/OnlinePlay/Components/StatusColouredContainer.cs
@@ -1,6 +1,7 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using System.ComponentModel;
 using osu.Framework.Allocation;
 using osu.Framework.Graphics;
@@ -30,7 +31,8 @@ protected override void LoadComplete()
 
             room.PropertyChanged += onRoomPropertyChanged;
 
-            Scheduler.AddDelayed(updateRoomStatus, 5000, true);
+            if (room.EndDate > DateTimeOffset.Now)
+                Scheduler.AddDelayed(updateRoomStatus, (room.EndDate.Value - DateTimeOffset.Now).TotalMilliseconds + 500, true);
             updateRoomStatus();
         }
 
diff --git a/osu.Game/Screens/OnlinePlay/Lounge/Components/RoomStatusPill.cs b/osu.Game/Screens/OnlinePlay/Lounge/Components/RoomStatusPill.cs
index 6da8f3ecbd..7007e6c8b8 100644
--- a/osu.Game/Screens/OnlinePlay/Lounge/Components/RoomStatusPill.cs
+++ b/osu.Game/Screens/OnlinePlay/Lounge/Components/RoomStatusPill.cs
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using System.ComponentModel;
 using osu.Framework.Allocation;
 using osu.Framework.Graphics;
@@ -37,7 +38,8 @@ protected override void LoadComplete()
 
             room.PropertyChanged += onRoomPropertyChanged;
 
-            Scheduler.AddDelayed(updateDisplay, 5000, true);
+            if (room.EndDate > DateTimeOffset.Now)
+                Scheduler.AddDelayed(updateDisplay, (room.EndDate.Value - DateTimeOffset.Now).TotalMilliseconds + 500, true);
             updateDisplay();
             FinishTransforms(true);
         }

```

It would at least mean that various elements would have their state changes synced up closer than `rand(0,5000)`.

Reply:
Hmm, I guess not since `RoomStatusPill` is also showing things like "is playing"?